prometheus/src/main/scala/filodb/prometheus/query/PrometheusModel.scala

```diff
@@ -84,16 +84,20 @@ object PrometheusModel {
                 (if (verbose) Map("_shards_" -> srv.key.sourceShards.mkString(","),
                                   "_partIds_" -> srv.key.partIds.mkString(","))
                 else Map.empty)
+    val samples = srv.rows.filter(!_.getDouble(1).isNaN).map { r =>
```

Comment:
Why is this needed?
Also you need to check the output schema.  Not all the outputs have double as the output type, and this code would crash.

Reply:
As discussed I made the change because without it Result never has None.
So when there are no rows in result it returns rangevectorkeys without any rows.